ccmlib/node.py

```diff
@@ -1986,12 +1986,22 @@ def log_line_category(line):
     loglines = log.splitlines()
 
     for line_num, line in enumerate(loglines):
-        if log_line_category(line) == 'ERROR':
+        found_exception = False
+        line_category = log_line_category(line)
+        if line_category == 'ERROR':
             matches.append([line])
+            found_exception = True
 
+        elif line_category == 'WARN':
+            match = re.search(r'exception', line)
```

Comment:
Woof, it totally does work like that. Could you make that `if match is not None`? I think they're functionally identical, but it matches the API for re.search more identically